src/fsharp/FSharp.Core/local.fs

```diff
@@ -125,7 +125,42 @@ module internal List =
                 let cons = freshConsNoTail x
                 chooseToFreshConsTail cons f t
                 cons
-            
+                  
+    let inline groupBy (comparer:IEqualityComparer<'SafeKey>) (keyf:'T->'SafeKey) (getKey:'SafeKey->'Key) (list: 'T list) =
```

Comment:
@KevinRansom So, the original function was inlined, which is why I have this one inlined. If this function isn't inlined, then the inlining on the function in `list.fs` doesn't do anything and there is probably an over 10x performance hit in a simple benchmark I tried.